File: java/src/hashing/DirectAddressing.java
package hashing;

public class DirectAddressing {
    String[] phoneBook = new String[10000000];

    private int getInt(String number) {
        return Integer.parseInt(number.replaceAll("\\D", ""));
    }

    public void setName(String number, String name) {
        int idx = this.getInt(number);
        this.phoneBook[idx] = name;
    }

    public String getName(String number) {
        int idx = this.getInt(number);
        return this.phoneBook[idx];
    }
}

File: python/hashing/directAddressing.py
import re

class DirectAddressing:
    def __init__(self):
        self.phoneBook = [None] * 10000000

    def _getInt(self, number):
        return int(re.sub('\D', '', number))

    def setName(self, number, name):
        idx = self._getInt(number)
        self.phoneBook[idx] = name

    def getName(self, number):
        idx = self._getInt(number)
        return self.phoneBook[idx]
